modules/common/Riot.py

```diff
@@ -53,30 +53,37 @@ def run_riot(self, owl_file, dir_output, json_file, owl_jq):
         # Set JVM memory limits
         run_env = os.environ.copy()
         run_env["_JAVA_OPTIONS"] = "-Xms4096m -Xmx8192m"
+
         try:
-            with open(path_output, "wb") as json_output, \
-                    subprocess.Popen([self.riot_cmd, "--output", "JSON-LD", owl_file], env=run_env,
-                                     stdout=subprocess.PIPE,
-                                     stderr=subprocess.PIPE) as riot_process, \
-                    subprocess.Popen([self.jq_cmd, "-r", owl_jq], env=run_env,
-                                     stdin=riot_process.stdout,
-                                     stdout=subprocess.PIPE,
-                                     stderr=subprocess.PIPE) as jq_process:
-                json_output.write(jq_process.stdout.read())
-                logger.error(f"RIOT stderr -> '{riot_process.stderr.read()}'")
-                logger.error(f"JQ stderr -> '{jq_process.stderr.read()}'")
-        except EnvironmentError as e:
-            msg_err = "When running RIOT for OWL file '{}', "
-            "with destination path '{}' and JQ filter '{}', "
-            "the following error occurred: '{}'".format(owl_file, path_output, owl_jq, e)
-            logger.error(msg_err)
-            raise RiotException(msg_err)
-        except Exception as e:
-            msg_err = "When running RIOT for OWL file '{}', "
-            "with destination path '{}' and JQ filter '{}', "
-            "the following error occurred: '{}'".format(owl_file, path_output, owl_jq, e)
-            logger.error(msg_err)
-            raise RiotException(msg_err)
+            logger.info(f"running Riot on {owl_file}")
+            riot_output = subprocess.run([self.riot_cmd, "--output", "JSON-LD", owl_file], env=run_env, capture_output=True)
```

Comment:
Got it, You are right, I'll refactor it now.

Reply:
For piping the riot-jq tandem in a single command, I wrote this code:
`riot_jq_cmd = f"{self.riot_cmd} --output 'JSON-LD' {owl_file} | {self.jq_cmd} -r '{owl_jq}' > {path_output}"`
`riot_jq_process = subprocess.run(riot_jq_cmd, env=run_env, shell=True, capture_output=True, check=True)`
But the problem with this code is when the process runs out of memor, it will be killed while the return code still is zero(0) and we only can find a "Killed" phrase in the stderr.
So in this condition we just can check the stderr for "Killed" string and then raise an exception.
If it's acceptable I can do in this way.